### DISCUSSION TITLE
pip list - SSL: UNKNOWN_PROTOCOL

### DISCUSSION THREAD

**Discussion by MagnaFrisia90** [HCL-TECH-SOFTWARE/connections-automation#271] [Q&A]
Hello,
to solve https://github.com/HCL-TECH-SOFTWARE/connections-automation/issues/269 I tried to install ansible with "pip3 install ansible" but I get the following error:

    Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLError(1, '[SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:877)'),)) - skipping

Also "pip3 list" & "pip list" print the same error.
I am using a http_proxy. This squid receives the request - log entry:

    1.2.3.4 - - [01/Mar/2023:14:44:30 +0100] "NONE error:invalid-request HTTP/1.1" 400 342 NONE:HIER_NONE

I have already talked to the Proxy-Admin about this. No further information can be provided.
So basically pip seems to be sending a wrong/malformed HTTP(s) request.
Does anybody run his envoirnment behind a proxy ? Or experienced the same issue?
Any ideas how to solve this ?

Thanks in advance !

(Also asked question here https://github.com/pypa/pip/issues/11826)

**Comment by MagnaFrisia90** [comment]
With Alma9 this just works. There were big issues with pretty-old CentOS7 last year.
